src/planscape/datasets/services.py

```diff
@@ -122,9 +123,37 @@ def create_style(
         data=data,
         data_hash=data_hash,
     )
+    action.send(created_by, verb="created", action_object=style)
     return {"style": style, "possibly_exists": hash_already_exists}
 
 
+@transaction.atomic()
+def assign_style(
+    created_by: User,
+    style: Style,
+    datalayer: DataLayer,
+) -> DataLayerHasStyle:
+    previous_association = DataLayerHasStyle.objects.filter(
+        style=style, datalayer=datalayer
+    ).first()
+    if previous_association:
+        previous_association.delete()
```

Comment:
```suggestion
    try:
        previous_association = DataLayerHasStyle.objects.select_for_update().get(
            style=style, datalayer=datalayer
        )
        previous_association.delete()
    except DataLayerHasStyle.DoesNotExist:
        pass
```
Just a suggestion to avoid problems with race condition scenarios.